ionics_fits/models/laser_rabi.py

```diff
@@ -176,11 +186,11 @@ def _func(
                 * (1 - np.exp(-t / tau) * np.cos(W * t))
             )
 
-            initial_state = self.distribution_fun(self.n_max, **kwargs)
-            P_i = np.tile(initial_state, (omega.shape[0], omega.shape[1], 1))
-            P_e = np.sum(P_i * P_trans, axis=-1).squeeze()
+            P_fock_i = self.distribution_fun(self.n_max, **kwargs)
+            # Transition probability averaged over Fock state distribution
```

Comment:
I'm not sure this kind of change is worthwhile. I don't really object to them, but I also don't think they make things any better - at this micro-level it feels like a matter of personal preference so changing is a poor use of time.

e.g. I don't like `P_fock_i` since it's not the probability of fock state `i`, bur rather a vector of fock state probabilities. Better IMHO would be `initial_distribution` or similar.

I won't block merging on account of this, but let's try not to make changes for their own sake

Reply:
In my opinion, this change improves readability. Numpy's broadcasting rules are straightforward, whereas I had to look up what the `np.tile` function does exactly and how it's meant to be called. I didn't introduce the `_i` suffix, that was already there in the previous version. I didn't remove it because I thought you had a reason for adding it (I assumed it meant "initial"). But can take it out now if you don't like it.

Reply:
aah, I see - I had misunderstood the changes. Are you sure the two are equivalent here?

Reply:
Yes, I have checked that `_func` correctly calculates transition probabilities if `x[0]` (duration) and `x[1]` (detuning) are scalars or vectors, which are also the cases supported by the current implementation using the `np.tile` function. I don't expect it to work when `t` or `detuning` are 2D arrays, but this is not currently supported either. One way that functionality would break is if the order of axes for the arrays `t`, `detuning` and `omega` created by the `np.meshgrid` function were changed. In particular, the trailing axis needs to correspond to the values for different Fock states. But this would also break in the current implementation.